pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -136,6 +136,13 @@ type ServiceMonitorSpec struct {
 	//
 	// +optional
 	AttachMetadata *AttachMetadata `json:"attachMetadata,omitempty"`
+	// When defined, enforcedBodySizeLimit specifies a job level limit on the size
+	// of uncompressed response body that will be accepted by Prometheus.
+	//
+	// It requires Prometheus >= v2.28.0.
+	//
+	// +optional
+	EnforcedBodySizeLimit ByteSize `json:"enforcedBodySizeLimit,omitempty"`
```

Comment:
Since it's an optional field.
```suggestion
	EnforcedBodySizeLimit *ByteSize `json:"enforcedBodySizeLimit,omitempty"`
```

Reply:
I'm wondering with worth we add a validation such as 
`+kubebuilder:validation:MinLength=1`

Reply:
``ByteSize` has already some validation.